Wikipedia/Code/MWKDataStore.m

```diff
@@ -114,25 +122,173 @@ - (void)dealloc {
 }
 
 - (instancetype)init {
-    self = [self initWithDatabase:[YapDatabase sharedInstance]];
+    self = [self initWithContainerURL:[[NSFileManager defaultManager] wmf_containerURL]];
     return self;
 }
 
-- (instancetype)initWithDatabase:(YapDatabase *)database {
-    self = [self initWithDatabase:database legacyDataBasePath:[[MWKDataStore class] mainDataStorePath]];
-    return self;
+static pid_t currentPid() {
```

Comment:
Since this is static, is there an implicit requirement that only a single MWKDataStore be instantiated per process? If so, should we somehow enforce this through a singleton or some other mechanism?